web/vite.config.ts

```diff
@@ -10,6 +10,20 @@ export default defineConfig(() => ({
     svelte(),
     nodeResolve({
       extensions: ['.js', '.ts']
-    })
+    }),
+    {
+      enforce: 'pre' as const,
+      // Deploy section.
+      name: 'html-transform',
+      // Changing the favicon depending on the environment mode
+      transformIndexHtml: (html: string) => {
+        return html.replace(
+          '/main/icon.svg',
```

Comment:
It is not cool that we replace SVG icon but keep old ICO.

Also replace `A` to `A` looks like wasting resources.

I suggest:

```js
if (process.env.NODE_ENV === 'development' || process.env.STAGING) {
  return html
    .replace('<link rel="icon" href="/favicon.ico" sizes="32x32" />', '')
    .replace()
} else {
  return html
}
```

In this case for production we will not do anything. And for development we will also remove ICO file.

Reply:
I found that it works for development, but doesn’t work in staging (this is why you do not see icon in PR preview deployment).

1. In staging Vite replaces `/main/icon.svg` to something like `__VITE_ASSET__Vk_3wFWY__` before running `transformIndexHtml`.
2. Replacing path to `/main/campfire.svg` will not add `campfire.svg` to `/dist` (it works in development because in development Vite doesn’t create `dist/` and look to file-syste, on every HTTP request from browser).

What do I suggest:

1. Replace RegExp to:
   ```js
       .replace(
          /<link rel="icon" href="[^"]+" type="image\/svg\+xml" \/>/,
          '<link rel="icon" href="/campfire.svg" type="image/svg+xml" />'
        )
   ```
   
   This RegExp is ready for `__VITE_ASSET__Vk_3wFWY__` URL.
2. Move `compfire.svg` from `web/main` to `web/public`. In this case Vite will always put it to `dist/`.

Reply:
How you can check that staging build is working (do it really).

1. Run `STAGING=1 pnpm production`
2. Open browser and check favicon.